### ISSUE TITLE
Feature Request: Auto-Configuration Tachometer

### ISSUE THREAD

**Issue by raamdev** [wpsharks/comet-cache#434]
@jaswsinc writes...

> I was just looking over some design concepts, when I saw this and it gave me an idea.
> 
> ![http://static2.bigstockphoto.com/thumbs/0/7/4/large2/47001286.jpg](http://static2.bigstockphoto.com/thumbs/0/7/4/large2/47001286.jpg)
> 
> It might be cool at some point to have a sort of auto-configuration wizard type of thinga-ma-jig, where you could enter into that mode and just choose from options like these.
> - Fast
> - Faster
> - Fastest
> 
> Where ZenCache would change your settings a bit for each of those speed levels. Fast might be the defaults, where Faster would be the addition of the HTML compressor, and Fastest might be GZIP, mod_expires, HTML compressor, etc, etc. It might be implemented as a tachometer, or perhaps just a slider would do. When the setting is changed, ZC could provide a brief summary of the configuration that will be applied.
> 
> Quick mock-up.
> 
> ![mockup](https://cloud.githubusercontent.com/assets/1563559/5343834/4dc63f6e-7eb6-11e4-8328-e483559de793.png)

**Comment by jaswrks** [comment]
Here's a start on this.
http://jsfiddle.net/JasWSInc/gz3pc0yf/1/

**Comment by raamdev** [comment]
Copying some of my comments from https://github.com/websharks/zencache.com/issues/55#issuecomment-112926736 related to this issue:

> I envisioned something that might be the perfect blend of what we're talking about. Right now we have a dual option of "Disable ZenCache / Enable ZenCache". What if we simply changed that to offer three options: "Disable ZenCache / Enable ZenCache (Safest Settings) / Enable ZenCache (Fastest Settings)"
> 
> ```
> [disable]               [enable-safest] [enable-fastest]
> ```
> 
> There could then be a brief description of those two settings, along with a note for "Experts" that they can choose either one and then go in and manually customize any of the settings they want.
> 
> Then whenever disabling/enabling the plugin on a site where there are already settings saved, we'd just provide a little prompt that says "Would you like to reset your configuration to the Safest settings?" (or "Fastest settings" if they chose `enable-fastest`). 

Do you still think we should go with a visual element like what you mocked up? My feeling is that we should keep this as straight-forward as possible and having fewer options (safest/fastest) makes the decision easier for site owners ("should I choose fast? faster? fastest? what's the difference? I want the fastest site, why can't I have that?").

**Comment by jaswrks** [comment]
> Do you still think we should go with a visual element like what you mocked up?

No, not necessarily. I'm following your lead on how this should work, and I'm absolutely OK with keeping it simple. I tossed that together after stumbling into work that was done here: http://www.code-front.com/apps/tachometer/

I was just getting a feel for how that library works and playing a bit. It might come in handy at some point in the future. I think visuals would help set ZenCache apart a bit, and everyone loves visual aids and fancy graphics w/ interaction. That also helps to give it a polished feel.

Whether we use it or not, I dunno. It's an interesting concept though.
